### PR TITLE
On cache clear

### PR DESCRIPTION
**Pull request by Squair**
This should satisfy #2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-cache-buster",
-  "version": "0.1.5",
```
Comment (CagriAldemir) [inline package.json:3]: Can u remove version increment line? Because I'm gonna add one more feature with this PR.

**Review by CagriAldemir — CHANGES_REQUESTED** [review]
Thank you for your support @Squair. I would be happy if you check my review comment.

**Comment by Squair** [comment]
@CagriAldemir Just noticed you made these changes yourself in master, just thought I'd double check - if you call onCacheClear before calling refreshCacheAndReload and it throws, wouldn't this give unpredictable results in the consuming code?

**Comment by CagriAldemir** [comment]
Actually, you're right. However, if I call the function later, the page will already be refreshed this time. Maybe it would be more correct to use the "onCacheClear" function as follows. What do you say?

```javascript
  const refreshCacheAndReload = async () => {
    try {
      if (window?.caches) {
        const { caches } = window;
        const cacheNames = await caches.keys();
        for (const cacheName of cacheNames) {
          caches.delete(cacheName);
        }
        log('The cache has been deleted.');
        onCacheClear && onCacheClear();
        window.location.reload(true);
      }
    } catch (error) {
      log('An error occurred while deleting the cache.', true);
      log(error, true);
    }
  };
```

**Comment by Squair** [comment]
@CagriAldemir It's better, but now I'm thinking about it, I'm not so sure about the practically of it. I don't think it's possible without getting a bit hacky, but it'd be really nice to be able to run onCacheClear after reloading the window, this way when someone hits an application you'd have a way to determine that the cache had to be cleared. This could then be used to notify them that new features were available or something like that - still possible before reloading I suppose, but a little more limiting

**Comment by CagriAldemir** [comment]
> @CagriAldemir It's better, but now I'm thinking about it, I'm not so sure about the practically of it. I don't think it's possible without getting a bit hacky, but it'd be really nice to be able to run onCacheClear after reloading the window, this way when someone hits an application you'd have a way to determine that the cache had to be cleared. This could then be used to notify them that new features were available or something like that - still possible before reloading I suppose, but a little more limiting

I agree with you. But to implement your idea, we have to store a flag in local storage or session storage. What do u think about it?

**Comment by Squair** [comment]
Yeah that could work - local storage feels like the best place. Happy to review a PR if you want a second pair of eyes 👍

**Pull request closed without merging**